packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import {Skeleton, Text, useMultiStyleConfig} from '@chakra-ui/react'
+import {useIntl} from 'react-intl'
+import {useCurrency} from '../../hooks'
+import useProductPricing from '../../hooks/use-product-pricing'
+
+const DisplayPrice = ({product, scope = 'pdp', quantity = 1}) => {
+    const intl = useIntl()
+    const {currency: activeCurrency} = useCurrency()
+    const {currency, basePrice, discountPrice} = useProductPricing(product, quantity)
+    const styles = useMultiStyleConfig('DisplayPrice', {variant: scope})
+    const isProductASet = product?.type?.set || product?.hitType === 'set'
+    const showDiscount = discountPrice !== null && discountPrice < basePrice
+    const basePriceStyle = showDiscount ? styles.discountPrice : styles.price
```

Comment:
nit: I would assume that `styles.discountPrice` is the styles for the discounted price, when it exists, not the base price when a discount applies. Something like `styles.basePriceWhenDiscounted` would be more clear. (But not necessarily exactly that - it feels a bit kludgy. Naming things is hard!)